### DISCUSSION TITLE
Basic authentication in a destination

### DISCUSSION THREAD

**Discussion by vincenzo-antolini-os** [dotnet/yarp#1878] [General] [answered]
I need an hint on a use case i faced today.

I can access a "origini website" throug a forward/client proxy, the curl command i use is:
`curl -x s1.proxymaster.com:10000 -U "username:password" https://basicauthenticatedhost.com`

I would like to setup a YARP to access the "origini website" without authenticating, this will make my workflow simpler.
Do you have some hint?

Thanks

**Comment by Tratcher** [comment, marked as answer]
-x is for a forward/client proxy. YARP is a reverse proxy and can rarely be used for the same scenarios. E.g. when the client connects to YARP it can't specify where requests are proxied to, that's the admin's decision.

YARP can add headers to a request. See https://microsoft.github.io/reverse-proxy/articles/transforms.html#requestheader. You could also do this in middleware: https://microsoft.github.io/reverse-proxy/articles/middleware.html
